### ISSUE TITLE
32-bit color issues on STM32F746G-DISCO

### ISSUE THREAD

**Issue by Simof92** [lvgl/lvgl#707]
Hi,

I'm trying to implement the demo application on the development board DISCO-STM32F746.
Screen 480x272 4.3"
Internal RAM 64KB
External SDRAM 64Mbit

The demo code work with 16bit color depth with the correct size of screen but the color of button is wrong (Horizontal Blu Lines)

I want to use 32bit color depth, I have configured the lv_conf.h and adapt the lcd driver with ARGB8888. Now the colors are correct but the image on the screen is divide in 4 equal parts and I don't understand what is the problem.

The VDB size is 480*272 and is allocated in External SDRAM.
![20190105_121531](https://user-images.githubusercontent.com/46396900/50723876-f7a64380-10e3-11e9-9ccd-e5f402949475.jpg)

![20190105_113905](https://user-images.githubusercontent.com/46396900/50723796-a8134800-10e2-11e9-9d10-b806eb81b7de.jpg)

[C_File.zip](https://github.com/littlevgl/lvgl/files/2729445/C_File.zip)

**Comment by embeddedt** [comment]
@Simof92 What a lucky coincidence! I just implemented 32-bit display support for that board earlier. Now you should just be able to set `LV_COLOR_DEPTH` to 32 and everything should work! Or is that what you already tried?

**Comment by Simof92** [comment]
With LV_COLOR_DEPTH to 16 the colors of the button are wrong (Horizontal Blue Lines) (Image 1)
With LV_COLOR_DEPTH to 32 the colors are correct but the size of screen is reduced (Image 2)

**Comment by embeddedt** [comment]
@Simof92 Please run `git pull` and `git submodule update`. I think you may not have my latest changes.

If it still doesn't work please send me the latest commit hash that you have.

**Comment by Simof92** [comment]
@embeddedt I have updated my Atollic project with your last changes and now the colors are perfect in 16bit depth. What was the problem in the previus version? Thank you a lot.
Now I will try with 32bit depth!

**Comment by embeddedt** [comment]
@Simof92 I'm not sure what your problem was, 16-bit color has always worked as far as I know. Maybe you happened to have an old version of the project.

**Comment by Simof92** [comment]
`LV_VDB_ADR (SDRAM_ADDR)+(LV_HOR_RES*LV_VER_RES*2)`
`LV_COLOR_DEPTH 32` 
DMA2D configured for ARGB8888
LTDC configured for ARGB8888

and the result is the same of the second picture.

The screen area is divide in 4 parts, in the first 2 parts is draw only half screen and in the other parts there are only noise

**Comment by embeddedt** [comment]
@Simof92 If you use the default `lv_conf.h` supplied with the F746 project do you have problems?

**Comment by Simof92** [comment]
@embeddedt after the update i don't have problem with the default configuration but LV_COLOR_DEPTH is configured 16 bit. 
With the changes described above it doesn't work correctly

**Comment by embeddedt** [comment]
@Simof92 Use the default configuration but change `LV_COLOR_DEPTH` to 32. That should work (it is working for me).

**Comment by embeddedt** [comment]
@Simof92 My mistake! I had made the pull request but the changes weren't actually in `master` yet; they were in a separate branch! Please `git pull` and `git submodule update` again.

**Comment by Simof92** [comment]
Great! Wonderful! Now is perfect! My next step will be to change the tft with 800x480 display! thank you a lot!

**Comment by embeddedt** [comment]
@Simof92 Glad to hear it helped you!

**Comment by Simof92** [comment]
Can you update the .ioc file created by STMCubeMX?
I have some problem to recreate the same project in Atollic Studio.

**Comment by embeddedt** [comment]
@Simof92 I never changed any `.ioc` files but I think @kisvegabor created the original project so he probably knows more about it than I do.

**Comment by Simof92** [comment]
This is the problem, Do you have some idea about the problem?

![20190112_110741_hdr](https://user-images.githubusercontent.com/46396900/51072045-d0f38a00-165a-11e9-92b1-0f45b331ee94.jpg)
[STM32F746_TFTDriver_v2.zip](https://github.com/littlevgl/lvgl/files/2751891/STM32F746_TFTDriver_v2.zip)

**Comment by Simof92** [comment]
The problem was in the SystemClockConfig function because it initialized the LTDC frequency

**Comment by kisvegabor** [comment]
@Simof92 STM32F746G project was created by someone else but I don't remember who. Anyway glad to hear that you solved the issue.
I was thinking about improving how the projects are added to LittlevGL. See [this](https://github.com/littlevgl/lvgl/issues/543#issuecomment-453749574).
It would be easier to see who created a project.

**Comment by Simof92** [comment]
I was able to rebuild the correct configuration in STMCubeMx and now everything works correctly in Atollic TrueSTUDIO.
I can upload the whole project if it will help anyone

**Comment by kisvegabor** [comment]
Great! Let's wait what @embeddedt thinks about this new project organization. If he also likes it I create a repo and give you write access there.

**Comment by Simof92** [comment]
Thanks for everything

**Comment by embeddedt** [comment]
@kisvegabor It's a much better solution and there will be less sub modules per project (and thus a smaller download size).

**Comment by kisvegabor** [comment]
I can imagine 3 solutions:
1. use 2 separate repos -> results in a lot of repos, and need to keep up to date both repos
2. use 1 repo with two folders for the 2 project -> need to update both repos
3. use 1 repo with two folders for the 2 project but they use the same code base -> might be difficult to copy a project

I vote for the 3rd option. What do you think?

**Comment by embeddedt** [comment]
@kisvegabor I've never used Atollic. But if it were possible to keep the IDE-specific stuff in two different directories that would be the simplest way.

@Simof92 Do you think this is possible?

**Comment by kisvegabor** [comment]
@embeddedt Atollic is also Eclipse-based and it has the same project settings to add source files.
So I think it's possible

**Comment by Simof92** [comment]
I don't now well how work the repos.
Atollic allows to work in different workspace

**Comment by kisvegabor** [comment]
I've imagined the structure like this:
- stm32f746_disco_no_os
  - atollic
    - main.c
    - other project realted things
  - sw4stm32
    - main.c
    - other project realted things
  - lvgl
  - lv_examples
  - hal_stm_lvgl

So you only need to link some external folders (lvgl, lv_eamples, hal_stm_lvgl) to the project.

You can do it in the project properties in Atollic and SW4STM32 too:
![image](https://user-images.githubusercontent.com/7599318/51076766-3cf6e200-169d-11e9-8111-55f1c5749367.png)

**Comment by embeddedt** [comment]
@kisvegabor Perfect!

**Comment by kisvegabor** [comment]
Awesome! :)

**Comment by kisvegabor** [comment]
I played with the "Link folder"  thing for a while and it's not so good as it seemed.
- more complicated project structure 
- more complex project settings (might be hard to follow for other projects)
- more complicated workspace structure because you will have an unused project.

In the light of it'd rather prefer separate repos for every IDE. 
I rename the current repo to `..._sw4stm32` and the create a new with `..._atollic`. Is it ok for you?

**Comment by kisvegabor** [comment]
I moved the f429 disco project: https://github.com/littlevgl/stm32f429_disco_no_os_sw4stm32

It was very easy to move the project this way.

**Comment by Simof92** [comment]
Yes i think is the better solution, when you will create the stm32f746_disco_no_os_atollic repo I will commit the project

**Comment by embeddedt** [comment]
@kisvegabor It is a better solution  in my opinion as well.

**Comment by kisvegabor** [comment]
Great!

Here are the repos:
- Atollic: https://github.com/littlevgl/stm32f746_disco_no_os_atollic 
- SW4STM32: https://github.com/littlevgl/stm32f746_disco_no_os_sw4stm32

**Comment by embeddedt** [comment]
@kisvegabor I've uploaded a working project to `stm32f746_disco_no_os_sw4stm32`.

**Comment by kisvegabor** [comment]
@embeddedt  Great thank you! I've added the git command to the README to clone the repo and the submodules.

**Comment by kisvegabor** [comment]
Awesome! Thank you, guys!

**Comment by seyyah** [comment]
Thanks a lot
